src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,388 @@
+map `double_centralizer.prod_mk := λ a, (a.left, a.right) : 𝓜(𝕜, A) → (A →L[𝕜] A) × (A →L[𝕜] A)`.
+We use this map to pull back the normed space structure from `(A →L[𝕜] A) × (A →L[𝕜] A)` to
+`𝓜(𝕜, A)`, which provides a definitional isometric embedding. Consequently, completeness of
+`𝓜(𝕜, A)` is obtained by proving that the range of this map is closed.
+-/
+
+namespace double_centralizer
+
+/-- the canonical map of `𝓜(𝕜, A)` into `(A →L[𝕜] A) × (A →L[𝕜] A)`. -/
+def prod_mk (a : 𝓜(𝕜, A)) : (A →L[𝕜] A) × (A →L[𝕜] A) := (a.left, a.right)
+
+@[simp] lemma prod_mk_def (a : 𝓜(𝕜, A)) : prod_mk 𝕜 A a = (a.left, a.right) := rfl
+
+variables {𝕜 A}
+
+lemma injective_prod_mk : function.injective (prod_mk 𝕜 A) :=
+λ a b h, ext a b (prod.ext_iff.mp h).1 (prod.ext_iff.mp h).2
+
+lemma range_prod_mk : set.range (prod_mk 𝕜 A) = {lr | ∀ x y, lr.2 x * y = x * lr.1 y} :=
+set.ext $ λ x, ⟨by {rintro ⟨a, rfl⟩, exact a.central}, λ hx, ⟨⟨x.1, x.2, hx⟩, by simp⟩⟩
+
+instance : has_add 𝓜(𝕜, A) :=
+{ add := λ a b,
+  { left := a.left + b.left,
+    right := a.right + b.right,
+    central := λ x y, by simp only [continuous_linear_map.add_apply, add_mul, mul_add, central] } }
+
+instance : has_zero 𝓜(𝕜, A) :=
+{ zero :=
+  { left := 0,
+    right := 0,
+    central := λ x y, by simp only [continuous_linear_map.zero_apply, zero_mul, mul_zero] } }
+
+instance : has_neg 𝓜(𝕜, A) :=
+{ neg := λ a,
+  { left := -(a.left),
+    right := -(a.right),
+    central := λ x y, by simp only [continuous_linear_map.neg_apply, neg_mul,
+                      mul_neg, neg_inj, central]}}
+
+instance : has_sub 𝓜(𝕜, A) :=
+{ sub := λ a b,
+  { left := a.left - b.left,
+    right := a.right - b.right,
+    central := λ x y, by simp only [continuous_linear_map.coe_sub', pi.sub_apply, sub_mul,
+      mul_sub, central] } }
+
+instance : has_smul 𝕜 𝓜(𝕜, A) :=
+{ smul := λ k a,
+  { left := k • a.left,
+    right := k • a.right,
+    central := λ x y , by simp only [continuous_linear_map.coe_smul', pi.smul_apply, central,
+      mul_smul_comm, smul_mul_assoc] } }
+
+@[simp] lemma add_left (a b : 𝓜(𝕜, A)) : (a + b).left = a.left + b.left := rfl
+@[simp] lemma add_right (a b : 𝓜(𝕜, A)) : (a + b).right = a.right + b.right := rfl
+@[simp] lemma zero_left : (0 : 𝓜(𝕜, A)).left = 0 := rfl
+@[simp] lemma zero_right : (0 : 𝓜(𝕜, A)).right = 0 := rfl
+@[simp] lemma neg_left (a : 𝓜(𝕜, A)) : (-a).left = -a.left := rfl
+@[simp] lemma neg_right (a : 𝓜(𝕜, A)) : (-a).right = -a.right := rfl
+@[simp] lemma sub_left (a b : 𝓜(𝕜, A)) : (a - b).left = a.left - b.left := rfl
+@[simp] lemma sub_right (a b : 𝓜(𝕜, A)) : (a - b).right = a.right - b.right := rfl
+@[simp] lemma smul_left (k : 𝕜) (a : 𝓜(𝕜, A)) : (k • a).left = k • a.left := rfl
+@[simp] lemma smul_right (k : 𝕜) (a : 𝓜(𝕜, A)) : (k • a).right = k • a.right := rfl
+
+-- this is easier than defining the instances of `has_smul` for `ℕ` and `ℤ` and pulling the group
```

Comment:
Arguably you should generalize the smul instance above to cover these cases; because by not doing so, you now don't have any lemmas that work for `(n • a).left`